pkg/network/render.go

```diff
@@ -389,3 +393,48 @@ func RenderMultusAdmissionController(conf *operv1.NetworkSpec, manifestDir strin
 	out = append(out, objs...)
 	return out, nil
 }
+
+func ClearNetAttachDef(client cli.Client, prev *operv1.NetworkSpec, conf *operv1.NetworkSpec) error {
+	prevNetAttachDefs := prev.AdditionalNetworks
+	curNetAttachDefs := conf.AdditionalNetworks
+	deleteNetAttachDefs := []operv1.AdditionalNetworkDefinition{}
+
+	for _, prev := range prevNetAttachDefs {
+		isFound := false
+		for _, cur := range curNetAttachDefs {
+			if prev.Name == cur.Name && prev.Namespace == cur.Namespace {
+				isFound = true
+				break
+			}
+		}
+
+		if isFound == false {
+			deleteNetAttachDefs = append(deleteNetAttachDefs, prev)
+		}
+	}
+
+	for _, netAttach := range deleteNetAttachDefs {
+		log.Printf("delete net-attach-def/%s namespace: %s", netAttach.Name, netAttach.Namespace)
+
+		if netAttach.Namespace == "" {
+			netAttach.Namespace = "default"
+		}
+
+		obj := &netattachv1.NetworkAttachmentDefinition{}
+		err := client.Get(context.TODO(), cli.ObjectKey{
+			Namespace: netAttach.Namespace,
+			Name:      netAttach.Name,
+		}, obj)
+		if err != nil {
+			log.Printf("failed to get net-attach-def %s in namespace %s: %v", netAttach.Name, netAttach.Namespace, err)
+			break
+		}
+
```

Comment:
You should only delete the NetAttachDef if the OwnerReference is set to ours. That way we won't accidentally step on anyone who re-creates it.